playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/KitsNavComponent.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState, useEffect, useContext } from "react"
+import { linkFormat } from "../../../../../../utilities/website_sidebar_helper"
+import RoutedNavItem from "../RoutedNavItem"
+import { useParams } from "react-router-dom"
+import { CollapsibleProvider } from "../CollapsibleContext"
+
+export const KitsNavItem = ({
+  link,
+  kitIndex,
+  collapsibles,
+  category,
+  dark,
+  isActive,
+  setIsActive,
+  updateTopLevelNav,
+  parentIndex,
+}) => {
+  const { name, type = "react" } = useParams()
+
+  const [collapsed] = collapsibles[kitIndex]
+
+  //make sure kits nav will stay toggled open when nested item is clicked
+  const updateKitsNav = (index) => {
+    collapsibles.forEach((collapsible, i) => {
+      const [, , setCollapsed] = collapsible
+      if (i !== index) {
+        setCollapsed(true)
+      } else {
+        setCollapsed(false)
+      }
+    })
+  }
+
+  //click on nested items
+  const handleSubItemClick = (subLinkIndex, sublink, Index) => {
+    updateTopLevelNav(parentIndex)
+    updateKitsNav(Index)
+    collapsibles[Index][2](false) // Use the setCollapsed function
+  }
+
+  if (typeof link === "object") {
```

Comment:
This is one of those areas that Sandy showed us how dependency injection can help clarify the contract of a class (React component in this case) so that the results are more predictable. If we have to check for type with `typeof` that is fine, but consider doing that beforehand and perhaps passing in the `RoutedNavItem` as the dependency.

Reply:
Gonna leave this for now, and wrestle with it in an upcoming story. Currently, none of the files inside of the Sidebar folder are being used in the beta view yet. 
I'll keep this note though for when we do try to implement this work.